### PR TITLE
feat(logger): check if terminal supports colors as default value

### PR DESCRIPTION
**Pull request by koddsson**
This PR uses the `supports-color` module to determine if the terminal that is running karma supports colors and sets the initial value of `useColors` to what `supports-color` determines. This value was previously hard-coded to `true`.

Users can still overwrite the setting to whatever they want in their config and the logger will respect that config value.

### CODE REVIEW

**Comment by koddsson** [comment]
@googlebot I consent.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
+<a name="4.3.0"></a>
+# [4.3.0](https://github.com/karma-runner/karma/compare/v4.2.0...v4.3.0) (2019-08-27)
+
+
+### Bug Fixes
+
```
Comment (johnjbarton) [inline CHANGELOG.md]: This file should not be part of the diff

File: package.json [hunk 2]
```diff
@@ -473,7 +474,7 @@
   "engines": {
     "node": ">= 8"
   },
-  "version": "4.2.0",
+  "version": "4.3.0",
```
Comment (johnjbarton) [inline package.json]: This change should not be part of the diff

**Comment by koddsson** [comment]
> Thanks, please rebase or otherwise clean up the diff.

All the other changes keep getting added by the pre-commit hook so I assumed they were supposed to go with the change.

**Comment by johnjbarton** [comment]
Sorry for the delay I had to fix appveyor. 

I think you need to 
$git rebase --hard HEAD~2 # remove dcce324ec88e5a434c4bffc774470e081db1ca2a and 40a8f07eb5b84ed7fb0adc162eb673e32035dcfa

$git checkout master
$git pull upstream master
$git checkout no-colors-when-not-supported
$git rebase master

This should put your change on top of the existing master with none of the extra commits.

**Comment by koddsson** [comment]
> Sorry for the delay I had to fix appveyor.

No worries 😺

**Comment by koddsson** [comment]
@johnjbarton Is there something more I should do here? No worries if not just figured I'd ask if there is something outstanding on my part 😸

**Comment by johnjbarton** [comment]
The CLA approval failed for some reason.

**Comment by koddsson** [comment]
@googlebot I consent.

**Comment by koddsson** [comment]
There seems to be some error happening @johnjbarton 😬. You can confirm my consent by just changing the label. 

> _Note to project maintainer:_ There may be cases where the author cannot leave a comment, or the comment is not properly detected as consent. In those cases, you can manually confirm consent of the commit author(s), and set the `cla` label to `yes` (if enabled on your project).

**Comment by jginsburgn** [comment]
Is this PR still relevant for you @koddsson ?

**Comment by koddsson** [comment]
@jginsburgn I'm ok with closing this.

**Comment by jginsburgn** [comment]
Ok. Thanks for updating this :)

**Pull request closed without merging**